src/methods/clipping/coverage.jl

```diff
@@ -0,0 +1,301 @@
+export coverage
+
+#=
+## What is coverage?
+
+Coverage is the amount of geometry area within a bounding box defined by the minimum and
+maximum x and y-coordiantes of that bounding box, or an Extent containing that information.
+
+To provide an example, consider this rectangle:
+```@example rect
+import GeometryOps as GO
+import GeoInterface as GI
+using Makie
+using CairoMakie
+
+rect = GI.Polygon([[(-1,0), (-1,1), (1,1), (1,0), (-1,0)]])
+cell = GI.Polygon([[(0, 0), (0, 2), (2, 2), (2, 0), (0, 0)]])
+xmin, xmax, ymin, ymax = 0, 2, 0, 2
+f, a, p = poly(collect(GI.getpoint(cell)); axis = (; aspect = DataAspect()))
+poly!(collect(GI.getpoint(rect)))
+f
+```
+It is clear that half of the polygon is within the cell, so the coverage should be 1.0, half
+of the area of the rectangle. 
+```@example rect
+GO.coverage(rect, xmin, xmax, ymin, ymax)
+```
+
+## Implementation
+
+This is the GeoInterface-compatible implementation. First, we implement a wrapper method
+that dispatches to the correct implementation based on the geometry trait. This is also used
+in the implementation, since it's a lot less work!
+
+Note that the coverage is zero for all points and curves, even if the curves are closed like
+with a linear ring.
+=#
+
+# Targets for applys functions
+const _COVERAGE_TARGETS = Union{GI.PolygonTrait,GI.AbstractCurveTrait,GI.MultiPointTrait,GI.PointTrait}
+
+# Wall types for coverage
+const UNKNOWN, NORTH, EAST, SOUTH, WEST = 0:4
+
+"""
+    coverage(geom, xmin, xmax, ymin, ymax, [T = Float64])::T
+
+Returns the area of intersection between given geometry and grid cell defined by its minimum
+and maximum x and y-values. This is computed differently for different geometries:
+
+- The signed area of a point is always zero.
+- The signed area of a curve is always zero.
+- The signed area of a polygon is calculated by tracing along its edges and switching to the
+    cell edges if needed.
+- The covverage of a geometry collection, multi-geometry, feature collection of
+    array/iterable is the sum of the coverages of all of the sub-geometries. 
+
+Result will be of type T, where T is an optional argument with a default value
+of Float64.
+"""
+function coverage(geom, xmin, xmax, ymin, ymax,::Type{T} = Float64; threaded=false) where T <: AbstractFloat
+    applyreduce(+, _COVERAGE_TARGETS, geom; threaded, init=zero(T)) do g
+        _coverage(T, GI.trait(g), g, T(xmin), T(xmax), T(ymin), T(ymax))
+    end
```

Comment:
This is the only thing I'm not sure about, since the function is inside the `applyreduce` call.  Still, it will operate at a pretty high level and hopefully that shouldn't cause too many issues.  

A potential solution might be to accept a tuple `(xmin, xmax, ymin, ymax)` in `_coverage`, and then just use `Base.Fix2` to create a function.

Reply:
Giving the anonymous function a name just on the line above can help the compiler. Surprisingly. 

All the T types we are using everywhere worry me a but too, the compiler likes to forget what they are... sometimes it can help to wrap them in `Val` further out to force them through

Reply:
I don't think I understand the problem. Can you elaborate? Is this something we need to fix pre-merge?

Reply:
This would be:
```suggestion
    coverage_for_subgeom(g) =  _coverage(T, GI.trait(g), g, T(xmin), T(xmax), T(ymin), T(ymax))
    applyreduce(coverage_for_subgeom, +, _COVERAGE_TARGETS, geom; threaded, init=zero(T))
```

which gets rid of the anonymous function created by the `do` block.

Reply:
Should I change this in `area` as well then, since I copied this format from there?

Reply:
I can do that in my PR